lib/bootloader_setup.pm

```diff
@@ -82,7 +82,7 @@ sub boot_into_snapshot {
     send_key_until_needlematch('boot-menu-snapshot', 'down', 10, 5);
     send_key 'ret';
     # assert needle to avoid send down key early in grub_test_snapshot.
-    assert_screen 'snap-default' if get_var('OFW');
+    assert_screen 'snap-default' if (get_var('OFW') && get_var('ROLLBACK_AFTER_MIGRATION'));
```

Comment:
I don't see why ROLLBACK_AFTER_MIGRATION should then ask for a screen check but not in other cases. The fix by @lemon-suse was valid in general and is not directly related to anything with migration. If there is a needle missing then it should be created. You can use existing needles based on this tag as a template. http://openqa.opensuse.org/admin/needles and the corresponding osd page should help